### ISSUE TITLE
Validation not working

### ISSUE THREAD

**Issue by noahlocke** [dcasia/conditional-container#26]
```php
  Number::make('Office', 'non_billable_doc_hours')->help('(non-billable)')->min(0)->max(1000)->step(0.25)->hideFromIndex(),
            ConditionalContainer::make([
                Text::make('Office Description', 'non_billable_doc_hours_description')->hideFromIndex()->rules('required'),
            ])->if('non_billable_doc_hours != 0'),
```

The intention here is that if I leave the `Office` field blank/zero, I can submit my form. If the "Office" field is not zero, then I show a text field and require input. Problem is, the form won't submit even if the "Office Description" is hidden, as I get a validation error stating that it's required. 

Am I mistaken, or should the "Office Description" field's validation rules only kick in if the field is shown?

**Comment by noahlocke** [comment]
OK I have an update, I figured out after reading #14 that this is due to the `!=` operator. Problem is, that's the only operator that will work with my `Number` field. I tried using:

```
->if('non_billable_doc_hours > 0')
```

In this case, the validation works as expected (field only validated when show), BUT I'm using the `step` method on my `Number`field and the conditional logic does not kick in until the value is  greater than `1`. So `0.25` will not trigger the conditional logic... 

Any ideas?
